client/azure_client_set.go

```diff
@@ -11,12 +11,194 @@ import (
 	"github.com/Azure/go-autorest/autorest"
 	"github.com/Azure/go-autorest/autorest/adal"
 	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/giantswarm/exporterkit/collector"
 	"github.com/giantswarm/microerror"
+	"github.com/prometheus/client_golang/prometheus"
 
 	"github.com/giantswarm/azure-operator/client/senddecorator"
 	"github.com/giantswarm/azure-operator/pkg/backpressure"
 )
 
+var (
+	// Prometheus metrics for clients
+	deploymentsClientMetricsDesc = prometheus.NewSummaryVec(
+		prometheus.SummaryOpts{
+			Name:       "azure_sdk_client_deployments",
+			Help:       "HTTP metrics for Azure SDK Deployments client.",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
```

Comment:
I need an ELI5 on that one. What is the mapping achieving? I can't really interpret it. Godoc of prometheus library was not really helpful on a brief glance.

Reply:
From https://godoc.org/github.com/prometheus/client_golang/prometheus#SummaryOpts:
```
    // Objectives defines the quantile rank estimates with their respective
    // absolute error. If Objectives[q] = e, then the value reported for q
    // will be the φ-quantile value for some φ between q-e and q+e.  The
    // default value is an empty map, resulting in a summary without
    // quantiles.
```

Reply:
I found that already, and then came here to ask for clarification. An example and reasoning would help. Neither the numbers nor the map tell me any story. I don't follow why they are selected as they are now.

Reply:
Why a summary? I'd have used a histogram

Reply:
> An example and reasoning would help. Neither the numbers nor the map tell me any story. I don't follow why they are selected as they are now.

50%, 90% and 99% percentiles are common defaults for these kind (like request latencies) of metrics. You can also grep for `quantile` from `g8s-grafana` - see e.g. https://github.com/giantswarm/g8s-grafana/blob/master/helm/g8s-grafana-chart/dashboards/shared/dns.json#L274-L294

> Why a summary? I'd have used a histogram.

Yeah, histogram would work as well. I guess we usually compute the quantiles on server-side. I was just looking into these metric types and since this readily calculates the quantile on client-side, it sort of made sense to me to do it that way.

Reply:
The fact that summaries calculates quantiles on client-side is why I tend towards histograms :D